mmdet/models/dense_heads/embedding_rpn_head.py

```diff
@@ -0,0 +1,124 @@
+import mmcv
+import torch
+import torch.nn as nn
+from mmcv import tensor2imgs
+
+from mmdet.models.builder import HEADS
+from ...core import bbox_cxcywh_to_xyxy
+
+
+@HEADS.register_module()
+class EmbeddingRPNHead(nn.Module):
+    """RPNHead in the `Sparse R-CNN <https://arxiv.org/abs/2011.12450>`_ .
+    Unlike traditional RPNHead, this module does not need FPN input, but just
+    decode init_proposal_bboxes and expand first dimension of
+    init_proposal_bboxes and init_proposal_features to the batch_size.
+
+    Args:
+        num_proposals (int): Number of init_proposals. Default 100.
+        proposal_feature_channel (int): Channel number of
+            init_proposal_feature
+        train_cfg (dict, optional): Necessary parameters of two-stage
+            detector, but always None in this module
+        test_cfg (dict, optional): Necessary parameters of two-stage
+            detector, but always None in this module
+    """
+
+    def __init__(
+        self,
+        num_proposals=100,
+        proposal_feature_channel=256,
+        train_cfg=None,
+        test_cfg=None,
+    ):
+        assert train_cfg is None
+        assert test_cfg is None
+        self.num_proposals = num_proposals
+        self.proposal_feature_channel = proposal_feature_channel
+        self.train_cfg = train_cfg
+        self.test_cfg = test_cfg
+        super(EmbeddingRPNHead, self).__init__()
+        self._init_layers()
+
+    def _init_layers(self):
+        """Initialize a sparse set of proposal boxes and proposal features."""
+        self.init_proposal_bboxes = nn.Embedding(self.num_proposals, 4)
+        self.init_proposal_features = nn.Embedding(
+            self.num_proposals, self.proposal_feature_channel)
+
+    def init_weights(self):
+        """Initialize the init_proposal_bboxes as normalized.
+
+        [c_x,c_y, w, h], and we initialize it to the size of  the entire image
+        """
+        nn.init.constant_(self.init_proposal_bboxes.weight[:, :2], 0.5)
```

Comment:
use constant_init in mmcv

Reply:
We use different initializations for different dimensions of weight, seems `mmcv.constant_init` don't support this.